pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,10 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// ExternalBuildStrategy defines input parameters specific to an External build.
+type ExternalBuildStrategy struct {
```

Comment:
The plan coming out of last week was to have Jenkins own a controller
watch/list loop reacting to new build configs.  It would invoke OS to
run image builds as necessary (using binary input) but would own the
whole flow.  So OS would handle creating the Jenkins instance on
demand, then the instance takes over managing them.

We definitely want a build config run by an external component, not
our controllers.  We want to decouple our controllers from knowing
details of builds, or having to launch builds to trigger external
build runs.  I think that's separating triggers from builds (by
mutation of the build config as some other work is going on).

If we can pin down exactly what data we record here and what the
Jenkins controller reports here, we're in a good spot to go ahead.  We
can use annotations first for status, but we have to forgive out how
to separate the controller now.